test/api/invite.test.js

```diff
@@ -1,70 +1,94 @@
+    version: 'v1',
+    groupKey: groupKey.toString('base64'),
+    root: groupInitMsg.key,
 
-      text: 'welcome friends',
-      recps: [groupId, ...authorIds],
+    text: 'welcome friends',
+    recps: [groupId, ...authorIds],
 
-      tangles: {
-        group: { root: groupInitMsg.key, previous: [groupInitMsg.key] },
-        members: { root: groupInitMsg.key, previous: [groupInitMsg.key] }
-      }
+    tangles: {
+      group: { root: groupInitMsg.key, previous: [groupInitMsg.key] },
+      members: { root: groupInitMsg.key, previous: [groupInitMsg.key] }
     }
-    t.deepEqual(invite.content, expected, 'kaitiaki sent invite')
+  }
+  t.deepEqual(invite.content, expected, 'kaitiaki sent invite')
 
-    /* kaitiaki posts to group, new person can read */
-    const greetingContent = {
-      type: 'post',
-      text: 'Welcome new person!',
-      recps: [groupId]
-    }
-    const { key: greetingKey } = await p(kaitiaki.publish)(greetingContent)
-    await p(replicate)({ from: kaitiaki, to: newPerson, live: false, name })
-    const greetingMsg = await p(Getter(newPerson))(greetingKey)
-    t.deepEqual(greetingMsg.value.content, greetingContent, 'new person can read group content')
-
-    /* new person posts to group, kaitiaki can read */
-    const replyContent = {
-      type: 'post',
-      text: 'Thank you kaitiaki',
-      recps: [groupId]
-    }
-    const { key: replyKey } = await p(newPerson.publish)(replyContent)
-    await p(replicate)({ from: newPerson, to: kaitiaki, live: false, name })
-    const replyMsg = await p(Getter(kaitiaki))(replyKey)
-    t.deepEqual(replyMsg.value.content, replyContent, 'kaitiaki can read things from new person')
-  } catch (err) {
-    t.fail(err)
+  /* kaitiaki posts to group, new person can read */
+  const greetingContent = {
+    type: 'post',
+    text: 'Welcome new person!',
+    recps: [groupId]
   }
+  const { key: greetingKey } = await run(
+    'kaitiaki published message',
+    p(kaitiaki.publish)(greetingContent)
+  )
+
+  let numberRebuilds = 0
+  const rebuildPromise = new Promise((resolve, reject) => {
+    // hook ssb.rebuild - this way we can piggyback the "done" callback of that
+    // and know when the get requests "should" work by
+    newPerson.rebuild.hook(function (rebuild, args) {
+      const [cb] = args
+      rebuild.call(this, (err) => {
+        numberRebuilds++
+        if (typeof cb === 'function') cb(err)
+        if (numberRebuilds === 1) resolve()
+      })
+    })
+  })
+
+  await run(
+    'replicated',
+    p(replicate)({ from: kaitiaki, to: newPerson, live: false })
+  )
+  await rebuildPromise
+
+  // const pull = require('pull-stream')
+  // const msgs = await pull(
+  //   newPerson.createLogStream({ private: true }),
+  //   pull.map(m => m.value.content),
+  //   pull.collectAsPromise()
+  // )
+  // console.log(msgs)
+
+  const greetingMsg = await run(
+    'new-person can get message',
+    Getter(newPerson)(greetingKey)
+  )
```

Comment:
@mixmix i think this was added as part of some of your debugging. do you want us to keep or remove it?

Reply:
keep it if it's not doing harm I guess